### PR TITLE
Make sure the test python versions are correct

### PR DESCRIPTION
**Pull request by jasongrout**
This makes sure that #3608 would have failed, since it shouldn't have worked with python 2. See #4174 and #4289.

### CODE REVIEW

**Comment by jasongrout** [comment]
Yep, sure enough, the test fails on master.

File: .travis.yml [hunk 1]
```diff
@@ -14,13 +14,18 @@ env:
   matrix:
     - GROUP=js
     - GROUP=js_cov
-    - GROUP=py3
-    - GROUP=py2
+    - GROUP=python PYTHON=2
+    - GROUP=python PYTHON=3.3
+    - GROUP=python PYTHON=3.4
+    - GROUP=python PYTHON=3.5
+    - GROUP=python PYTHON=3
```
Comment (ian-r-rose) [inline .travis.yml]: Will this explode our CI even more?

Comment (jasongrout) [reply .travis.yml]: Yes, unfortunately. Do you think it's worth it to test it?

Comment (jasongrout) [reply .travis.yml]: It adds three more runs of the python group, and one more run of the cli group.

Comment (ian-r-rose) [reply .travis.yml]: At the very least we can get rid of 3.3. IPython dropped support a week ago: https://github.com/ipython/ipython/pull/10833

Comment (ian-r-rose) [reply .travis.yml]: It's easy enough to ratchet this back if it gets unreasonable.

Comment (jasongrout) [reply .travis.yml]: Looks like 3.3 and 3.4 will have problems using the conda-forge packages. Here's the install error with 3.3:

```
UnsatisfiableError: The following specifications were found to be in conflict:
  - notebook -> ipykernel -> ipython[version='>=4.0'] -> pickleshare -> python=3.5 -> tk=8.6
  - notebook -> ipykernel -> ipython[version='>=4.0'] -> pickleshare -> python=3.5 -> xz[version='>=5.2.3,<6.0a0']
  - python=3.3
```
I'll remove 3.3 and 3.4 from the testing. That gives us the last two minor versions of 3 for testing.

**Comment by jasongrout** [comment]
I also merged master into this branch, so now the 2.7 tests should pass.

**Comment by jasongrout** [comment]
It looks like we have some other legitimate python 2 problems in semver.py. Investigating...

**Pull request merged**